app/src/main/java/co/minesweepers/birthday/activities/GetRecipientActivity.java

```diff
@@ -0,0 +1,117 @@
+package co.minesweepers.birthday.activities;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import co.minesweepers.birthday.R;
+import co.minesweepers.birthday.Utils;
+import co.minesweepers.birthday.model.Memory;
+
+public class GetRecipientActivity extends AppCompatActivity implements View.OnClickListener {
+
+    private EditText mNameTxt;
+    private ImageView mNameContinueBtn;
+    private TextView mUploadImageText;
+    private TextView mRecipientNameInfo;
+    private ImageView mUploadImageView;
+    private Button mButtonContinue;
+    private Memory mMemory;
+
+    private static final int PICK_IMAGE_REQUEST = 100;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_get_recipient);
+        mMemory = Memory.getInstance();
+        mNameContinueBtn = (ImageView) findViewById(R.id.recipient_name_continue);
+        mNameContinueBtn.bringToFront();
+        mNameContinueBtn.setOnClickListener(this);
+        mNameTxt = (EditText) findViewById(R.id.recipient_name);
+        mRecipientNameInfo = (TextView) findViewById(R.id.recipient_name_info);
+        mButtonContinue = (Button) findViewById(R.id.button_recipient_done);
+        mButtonContinue.setOnClickListener(this);
+        mUploadImageText = (TextView) findViewById(R.id.get_recipient_upload_pic_info);
+        mUploadImageView = (ImageView) findViewById(R.id.get_recipient_upload_pic);
+        mUploadImageView.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
+            case R.id.recipient_name_continue:
+                if (Utils.isEmpty(mNameTxt.getText().toString())) {
+                    Toast.makeText(getApplicationContext(), R.string.recipient_name_empty, Toast.LENGTH_SHORT).show();
+                } else {
+                    closeKeyboard();
+
+                    mUploadImageText.setVisibility(View.VISIBLE);
+                    mUploadImageView.setVisibility(View.VISIBLE);
+                    mNameContinueBtn.setVisibility(View.INVISIBLE);
+
+                    mNameTxt.setFocusable(false);
+                    mNameTxt.setClickable(false);
+                }
+                break;
+
+            case R.id.get_recipient_upload_pic:
+                showPictureChooser();
+                break;
+
+            case R.id.button_recipient_done:
+                Intent viewMemoryIntent = new Intent(this, AddPeopleActivity.class);
```

Comment:
look at line 106 below . Continue button is not visible to the user till mMemory.setRecipient triggers onSuccess callback . User won't see this button ( and won't go to AddPeopleActivity ) without setting a recipient .